src/lib.rs

```diff
@@ -442,26 +447,27 @@ impl MeasurementDuration for sensor_class::ShtGeneric {
 /// [`ShtCx`]: struct.ShtCx.html
 /// [`MeasurementDuration`]: trait.MeasurementDuration.html
 #[inline(always)]
-pub fn max_measurement_duration<S, I2C>(_: &ShtCx<S, I2C>, mode: PowerMode) -> u16
+pub fn max_measurement_duration<S, I2C>(_: &ShtCx<S, I2C>, mode: PowerMode) -> u32
 where
     S: ShtSensor + MeasurementDuration,
 {
     S::max_measurement_duration(mode)
 }
 
 /// General functions.
-impl<S, I2C, E> ShtCx<S, I2C>
+impl<S, I2C> ShtCx<S, I2C>
 where
     S: ShtSensor,
-    I2C: Read<Error = E> + Write<Error = E>,
+    I2C: I2c<SevenBitAddress>,
+    I2C::Error: Into<Error<I2C::Error>>,
```

Comment:
Hm, if I understand this correctly, this means: The type can be instantiated using any I2C HAL implementation that implements `Into<Error<...>>`, with `Error` being the error type of this crate.

Instead of adding a trait bound on the HAL's error type, we should instead add a trait bound on our own error type (as suggested above).